apps/user-ms/src/main/java/it/pagopa/selfcare/user/entity/filter/OnboardedProductFilter.java

```diff
@@ -61,7 +61,8 @@ public Map<String, Object> constructMap() {
         map.put(CREATED_AT.getChild(), createdAt);
         map.put(UPDATED_AT.getChild(), updatedAt);
 
-        map.values().removeIf(Objects::isNull);
+        map.entrySet().removeIf(e -> Objects.isNull(e.getValue()) ||
+                (e.getValue() instanceof Collection && ((Collection) e.getValue()).isEmpty()));
```

Comment:
you could avoid to iterate the entries for removing item null using above for example **Optional.ofNullable(updatedAt).ifPresent(value -> map.put(...))**

Reply:
Anyway I should iterate entries to check if some input is a list and empty list